src/main/java/seedu/academydirectory/logic/commands/SortCommand.java

```diff
@@ -0,0 +1,123 @@
+package seedu.academydirectory.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.academydirectory.logic.parser.CliSyntax.PREFIX_ATTRIBUTE;
+import static seedu.academydirectory.logic.parser.CliSyntax.PREFIX_ORDER;
+import static seedu.academydirectory.model.Model.PREDICATE_SHOW_ALL_STUDENTS;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.academydirectory.model.AcademyDirectory;
+import seedu.academydirectory.model.Model;
+import seedu.academydirectory.model.student.Assessment;
+import seedu.academydirectory.model.student.Name;
+import seedu.academydirectory.model.student.Participation;
+import seedu.academydirectory.model.student.Student;
+
+/**
+ * A class that implements the command to show all the students' grades for an assessment.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String HELP_MESSAGE = "### Sorts students by specific attribute:  `sort`\n"
+            + "Avengers will be able to sort their AcademyDirectory by the specified attribute.\n"
+            + "\n"
+            + "Format: `sort attr/ATTRIBUTE ord/ORDER`\n"
+            + "\n"
+            + "Sorts AcademyDirectory in `ORDER` order by their `ATTRIBUTE`.\n"
+            + "The input `ATTRIBUTE` must be a valid attribute: name, participation,"
+            + " RA1, MIDTERM, RA2, PE, FINAL, AVERAGE.\n"
+            + "The input `ORDER` must be a valid order: asc, desc"
+            + "\n"
+            + "Example:\n"
+            + "* `sort attr/name /asc`"
+            + "* `sort attr/RA1 /desc`";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts student by the specified attribute. "
+            + "\nParameters: "
+            + PREFIX_ATTRIBUTE + "ATTRIBUTE "
+            + PREFIX_ORDER + " ORDER "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_ATTRIBUTE + "name "
+            + PREFIX_ORDER + "asc";
+
+    private final String attribute;
+    private final boolean isAscendingOrder;
+
+    /**
+     * Constructor for ShowCommand.
+     * @param attribute The attribute to be sorted by.
+     * @param isAscendingOrder The order in which to sort.
+     */
+    public SortCommand(String attribute, boolean isAscendingOrder) {
+        requireNonNull(attribute);
+        requireNonNull(isAscendingOrder);
+        this.attribute = attribute.toUpperCase();
+        this.isAscendingOrder = isAscendingOrder;
+    }
+
+    /**
+     * Return a String representation of the respective sort executed.
+     * @return A String representation of the sort executed.
+     */
+    public String displayResult() {
+        String result = "AcademyDirectory is now sorted by ";
+        String orderString = isAscendingOrder ? "ascending" : "descending";
+        return result + attribute + " in " + orderString + " order!";
+    }
```

Comment:
Personally think the name's misleading cuz it doesn't actually display but rather produce a displayable version

Reply:
wdyt about it being 'getResultString'?